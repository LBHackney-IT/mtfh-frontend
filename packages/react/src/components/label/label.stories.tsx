import React from "react"

import { Label } from "./label"

export default {
  title: "Label",
  component: Label,
}

export const Base = () => {
  return <Label htmlFor="test">A descriptive label</Label>
}

export const PolymorphicLabel = () => {
  return <Label as="p">A descriptive label</Label>
}
