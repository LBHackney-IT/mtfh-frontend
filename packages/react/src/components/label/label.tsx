import React, { forwardRef } from "react"
import type * as Polymorphic from "@radix-ui/react-polymorphic"
import cn from "classnames"

import styles from "./styles.module.scss"

export interface LabelProps {
  variant?: "base" | "sm" | "md" | "lg" | "xl"
}

export type LabelComponent = Polymorphic.ForwardRefComponent<
  "label",
  LabelProps
>

export const Label: LabelComponent = forwardRef(function Label(
  { as: LabelComp = "label", variant = "base", className, ...props },
  ref,
) {
  return (
    <LabelComp
      ref={ref}
      className={cn(
        styles.govukLabel,
        {
          [styles.govukLabelS]: variant === "sm",
          [styles.govukLabelM]: variant === "md",
          [styles.govukLabelL]: variant === "lg",
          [styles.govukLabelXL]: variant === "xl",
        },
        styles.lbhLabel,
        className,
      )}
      {...props}
    />
  )
})
