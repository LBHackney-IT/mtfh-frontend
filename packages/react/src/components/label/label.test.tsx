import React from "react";
import { render, testA11y } from "@hackney/mtfh-test-utils";

import { Label } from "./label";

test("it renders the label", async () => {
  const { container } = render(
    <>
      <Label htmlFor="input">Label</Label>
      <input id="input" />
    </>,
  );
  expect(container).toMatchSnapshot();
  await testA11y(container);
});
