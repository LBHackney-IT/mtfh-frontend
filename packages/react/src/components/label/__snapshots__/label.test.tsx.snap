// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the label 1`] = `
<div>
  <label
    class="govukLabel lbhLabel"
    for="input"
  >
    Label
  </label>
  <input
    id="input"
  />
</div>
`;
